prisma/schema.prisma

```diff
@@ -9,3 +9,65 @@ datasource db {
   provider = "postgresql"
   url      = env("DATABASE_URL")
 }
+
+model Post {
+  id           Int      @id @default(autoincrement())
+  fields       String[]
+  title        String
+  Authors      User[]
+  content      String
+  Links        Post[]   @relation(name: "Links")
+  Revisions    Post[]   @relation(name: "Revisions")
+  NextRevision Post?    @relation(name: "NextRevision")
+  sponsored    Boolean  @default(false)
+  release      DateTime @default(now()) @db.Date
+  score        Int      @default(0)
+
+  PostLinks      Post?       @relation("Links", fields: [postIdLink], references: [id])
+  PostRevision   Post?       @relation("Revisions", fields: [postIdRev], references: [id])
+  postIdLink     Int?
+  postIdRev      Int?
+  ScientificPub  Scientific? @relation("Publications", fields: [scienOrcidVote], references: [orcid])
+  ScientificVote Scientific? @relation("Vote", fields: [scienOrcidPub], references: [orcid])
+  scienOrcidPub  Int?
+  scienOrcidVote Int?
+  PrevRevision   Post?       @relation("NextRevision", fields: [PrevRevisionId], references: [id])
+  PrevRevisionId Int?        @unique
+}
```

Comment:
Does the model 'Post' link to itself sometimes?

Reply:
Yep but it's normal.
It allows to be able to link other related posts.
It is also for practical purpose about updating a post.